client/state/data-layer/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import {
+	isArray,
+	mergeWith,
+} from 'lodash';
+
+const concatHandlers = ( left, right ) =>
+	isArray( left )
+		? left.concat( right )
+		: undefined;
+
+export const mergeHandlers = ( ...handlers ) =>
+	handlers.length > 1
+		? mergeWith( {}, ...handlers, concatHandlers )
+		: handlers.shift();
```

Comment:
> What's the use case for passing a single handler to this function?

there is none other than this very PR when all we have is one but more are expected. I could have left it out but I knew we'd need it. I only wanted to add a single handler to make this PR small and I also wanted to be able to provide the example on which others can be built. it's a chicken-and-egg sort of thing because we'll need `mergeHandlers` on the next PR that builds off of this

> Would it be more reasonable to accept a single array argument instead of the spread arguments? What if we want to add an options argument in the future?

personally I didn't see much need to accept an array but it's obviously a trivial thing to do. can you think of any cases where we would need an options argument?